cryptolib/cryptolib.go

```diff
@@ -44,30 +52,41 @@ func GetQuoteForCryptoAsset(priceFeeds []*PriceFeed, asset string) (*messagelib.
 	return &messagelib.TickerValue{Ticker: asset, Value: float32(price), Change: float32(change) * 100.0}, nil
 }
 
-func getPriceFeed(priceFeeds []*PriceFeed, asset string) (*PriceFeed, bool) {
-	for i := range priceFeeds {
-		if priceFeeds[i].Pair == asset {
-			return priceFeeds[i], true
+func getFeedForAsset(geminiClient *http.Client, asset string) (*PriceFeed, bool) {
+	fetchPriceFeeds(geminiClient)
+	for _, feed := range priceFeeds {
+		if feed.Pair == asset {
+			return feed, true
 		}
 	}
 	return nil, false
 }
 
-// GetPriceFeeds returns the current Price points of cryptos traded on Gemini.
-func GetPriceFeeds(geminiClient *http.Client) ([]*PriceFeed, error) {
-	var priceFeeds []*PriceFeed
+func fetchPriceFeeds(geminiClient *http.Client) {
+	mu.Lock()
+	defer mu.Unlock()
```

Comment:
No Action: So this is how we handle the lazy loading right? The equivalent in java would be:

public synchronized PriceFeeds fetchPriceFeeds(Client client) {
  if time.Since(lastUpdated) <= priceFeedAgeLimit {
		return cachedPriceFeeds
	}
   cachePriceFeeds = actuallyFetch(client);
return cachedPriceFeeds;
}

Right?

Reply:
Yep.